js_modules/dagster-ui/packages/ui-core/src/runs/useJobReExecution.tsx

```diff
@@ -29,9 +37,41 @@ export const useJobReexecution = (opts?: {onCompleted?: () => void}) => {
 
   return useCallback(
     async (
-      run: {id: string; pipelineName: string},
+      run: {id: string; pipelineName: string; tags: {key: string; value: string}[]},
       param: ReexecutionStrategy | ExecutionParams,
     ) => {
+      const backfillTag = run.tags.find((t) => t.key === DagsterTag.Backfill);
+
+      if (backfillTag) {
+        const {data} = await client.query<
+          CheckBackfillStatusQuery,
+          CheckBackfillStatusQueryVariables
+        >({
+          query: CHECK_BACKFILL_STATUS_QUERY,
+          fetchPolicy: 'no-cache',
+          variables: {
+            backfillId: backfillTag.value,
+          },
+        });
+        const backfillRunning =
+          data.partitionBackfillOrError.__typename === 'PartitionBackfill'
+            ? data.partitionBackfillOrError.status === BulkActionStatus.REQUESTED
+            : false;
+
+        if (!backfillRunning) {
+          try {
+            await confirm({
+              title: 'Re-execution within backfill',
+              description: `This run is part of a completed backfill. Re-executing will not update the backfill status or launch runs of downstream dependencies.`,
+              intent: 'primary',
+              catchOnCancel: true,
```

Comment:
Necessary?

Reply:
Ahh yeah, it's interesting - the default behavior of this confirm is to just never resolve if the user clicks cancel. We actually need to throw, then catch here so we can return from this function and stop the button from spinning. Glad we have this option!